src/keyboard-layout-manager-mac.mm

```diff
@@ -47,20 +54,18 @@ static void RemoveCFObserver(void *arg) {
   node::AddEnvironmentCleanupHook(
     isolate,
     RemoveCFObserver,
-    const_cast<void*>(static_cast<const void*>(nullptr)));
+    const_cast<void*>(static_cast<const void*>(this)));
 #endif
 }
 
 KeyboardLayoutManager::~KeyboardLayoutManager() {
-#if NODE_MAJOR_VERSION >= 10
-  node::RemoveEnvironmentCleanupHook(
-    isolate(),
-    RemoveCFObserver,
-    const_cast<void*>(static_cast<const void*>(this))
-  );
-#endif
+#if NODE_MAJOR_VERSION < 10
   RemoveCFObserver(this);
+  uv_close(reinterpret_cast<uv_handle_t *>(async), [](uv_handle_t *handle){
+    free(handle);
+  });
   delete callback;
+#endif
```

Comment:
- Would it be possible to call KeyboardLayoutManager's constructor and destructor multiple times in the same context, so that node will keep adding CleanupHooks to its internal list but never removing them? if so, we should also call `node::RemoveEnvironmentCleanupHook` in the destructor.
- Also, for NODE_MAJOR_VERSION >= 10, `RemoveCFObserver(this);` and `delete callback` are not being called in the destructor.

Reply:
Code refactored